### PR TITLE
Gpu support

### PR DESCRIPTION
**Pull request by thecodejedi**
Download a Full FFMpeg binary package so that HW Encodings are supported by default

### CODE REVIEW

File: Xabe.FFmpeg.Test/AudioStreamTests.cs [hunk 1]
```diff
@@ -89,8 +89,8 @@ public async Task ChangeSamplerate()
             string outputPath = Path.ChangeExtension(Path.GetTempFileName(), FileExtensions.Mp3);
 
             var audioStream = inputFile.AudioStreams.First();
-            var sampleRate = audioStream.SampleRate;
-            Assert.Equal(48000, sampleRate);
+            var channels = audioStream.SampleRate;
```
Comment (lkischler) [inline Xabe.FFmpeg.Test/AudioStreamTests.cs]: Why? It's SampleRate not channels that I think

File: Xabe.FFmpeg.Test/DownloaderTests.cs [hunk 2]
```diff
@@ -18,26 +21,27 @@ internal async Task FullProcessPassed()
 
             var operatingSystemProvider = Substitute.For<IOperatingSystemProvider>();
             operatingSystemProvider.GetOperatingSystem().Returns(x => os);
-            FFmpegDownloader._linkProvider = new LinkProvider(operatingSystemProvider);
+            FFmpegDownloader downloader = new FFmpegDownloader();
+            downloader._linkProvider = new LinkProvider(operatingSystemProvider);
```
Comment (lkischler) [inline Xabe.FFmpeg.Test/DownloaderTests.cs]: It's should be property nammed LinkProvider with public getter

File: Xabe.FFmpeg/Downloader/FFMpegDownloaderBase.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            {
+                if (!Directory.Exists(destinationDir))
+                    Directory.CreateDirectory(destinationDir);
+
+                foreach (ZipArchiveEntry zipEntry in zipArchive.Entries)
+                {
+                    string destinationPath = Path.Combine(destinationDir, zipEntry.FullName);
+
+                    // Archived empty directories have empty Names
+                    if (zipEntry.Name == string.Empty)
```
Comment (lkischler) [inline Xabe.FFmpeg/Downloader/FFMpegDownloaderBase.cs]: Is not string.IsNullOrEmpty better choice?

File: Xabe.FFmpeg/Downloader/Official/FFmpegDownloader.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace Xabe.FFmpeg.Downloader.Official
+{
+    internal class FFmpegDownloader : FFmpegDownloaderBase
+    {
+        internal LinkProvider _linkProvider = new LinkProvider();
```
Comment (lkischler) [inline Xabe.FFmpeg/Downloader/Official/FFmpegDownloader.cs]: It should be property. Not field

File: Xabe.FFmpeg/FFmpeg.cs [hunk 5]
```diff
@@ -56,10 +58,20 @@ public abstract class FFmpeg
 
         /// <summary>
         ///     Download latest FFmpeg version for current operating system to FFmpeg.ExecutablePath. If it is not set download to ".".
+        /// <param id="gpuSupport">Determine if a fully compiled FFPmpeg is downloaded</param>
         /// </summary>
-        public static Task GetLatestVersion()
+        public static Task GetLatestVersion(bool gpuSupport = false)
```
Comment (lkischler) [inline Xabe.FFmpeg/FFmpeg.cs]: Probably GPU support is not only one feature in full version. Maybe this param should be named as 'fullVersion'. Description about differences in this two versions should be in comment.

**Comment by tomaszzmuda** [comment]
Merged in #188

**Pull request closed without merging**